src/Control/Carrier/Accum/Strict.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+{- | A carrier for 'Acccum' effects.
+This carrier performs its append operations strictly and thus avoids the space leaks inherent in lazy writer monads.
+These appends are left-associative; as such, @[]@ is a poor choice of monoid for computations that entail many calls to 'tell'.
+The [Seq](http://hackage.haskell.org/package/containersdocs/Data-Sequence.html) or [DList](http://hackage.haskell.org/package/dlist) monoids may be a superior choice.
+
+-- | @since 1.0.3.0 -- FIXME Is this right?
+-}
+
+module Control.Carrier.Accum.Strict
+( -- * Accum carrier
+  runAccum
+, execAccum
+, AccumC(..)
+  -- * Accum effect
+, module Control.Effect.Accum
+) where
+
+import Control.Algebra
+import Control.Applicative (Alternative)
+import Control.Carrier.State.Strict
+import Control.Effect.Accum
+import Control.Monad (MonadPlus)
+import Control.Monad.Fail as Fail
+import Control.Monad.Fix
+import Control.Monad.IO.Class
+import Control.Monad.Trans.Class
+
+-- | Run an 'Accum' effect with a 'Monoid'al log, applying a continuation to the final log and result.
+--
+-- @
+-- 'runAccum' k ('pure' a) = k 'mempty' a
+-- @
+-- @
+-- 'runAccum' k ('add' w) = k w ()
+-- @
+-- @
+-- 'runAccum' k ('add' w >> 'look') = k w w
+-- @
+--
+-- @since 1.0.3.0 -- FIXME Is this right?
+runAccum :: Monoid w => AccumC w m a -> m (w, a)
+runAccum = runState mempty . runAccumC
+{-# INLINE runAccum #-}
+
+-- | Run a 'Accum' effect with a 'Monoid'al log, producing the final log and discarding the result value.
+--
+-- @
+-- 'execAccum' = 'runAccum' ('const' '.' 'pure')
+-- @
+--
+-- @since 1.0.3.0 -- FIXME Is this right?
+execAccum :: (Monoid w, Applicative m) => AccumC w m a -> m w
+execAccum = fmap fst . runAccum
```

Comment:
Ditto here; `evalAccum` would be a welcome addition.

Reply:
Yes, will add.